modules/ducktests/tests/ignitetest/tests/two_phased_rebalanced_test.py

```diff
@@ -0,0 +1,294 @@
+
+        for item in res.items():
+            self.logger.info(f'Host: {item[0]}, PDS {item[1]}mb')
+
+        return res
+
+    def await_cluster_idle(self, node: ClusterNode):
+        """
+        Await Skipping checkpoint.
+        :param node ClusterNode.
+        """
+        try:
+            IgniteAwareService.await_event_on_node('Skipping checkpoint', node, timeout_sec=60)
+        except errors.TimeoutError as ex:
+            self.logger.warn(ex)
+
+
+def restart_with_clean_idx_node_on_cell_and_await_rebalance(cells: [IgniteService], idxs: [int]):
+    """
+    Restart idxs nodes on cells with cleaning working directory and await rebalance.
+    :param cells List of IgniteService.
+    :param idxs List the index nodes that need to be restarted with cleanup.
+    """
+    stop_idx_node_on_cell(cells, idxs)
+    clean_work_idx_node_on_cell(cells, idxs)
+    start_idx_node_on_cell(cells, idxs)
+
+    for cell in cells:
+        cell.await_rebalance()
+
+
+def stop_idx_node_on_cell(cells: [IgniteService], idxs: [int]):
+    """
+    Stop idxs nodes on cells.
+    :param cells List of IgniteService.
+    :param idxs List of index nodes to stop.
+    """
+    for cell in cells:
+        size = len(cell.nodes)
+
+        for i in idxs:
+            assert i < size
+
+            cell.stop_node(cell.nodes[i])
+
+    for cell in cells:
+        for i in idxs:
+            cell.wait_node(cell.nodes[i])
+
+
+def clean_work_idx_node_on_cell(cells: [IgniteService], idxs: [int]):
+    """
+    Cleaning the working directory on idxs nodes in cells.
+    :param cells List of IgniteService.
+    :param idxs List of index nodes to clean.
+    """
+    for cell in cells:
+        size = len(cell.nodes)
+
+        for i in idxs:
+            assert i < size
+
+            node = cell.nodes[i]
+
+            assert not cell.pids(node)
+
+            node.account.ssh(f'rm -rf {cell.work_dir}')
+
+
+def start_idx_node_on_cell(cells: [IgniteService], idxs: [int]):
+    """
+    Start idxs nodes on cells.
+    :param cells List of IgniteService.
+    :param idxs List of index nodes to start.
+    """
+    for cell in cells:
+        size = len(cell.nodes)
+
+        for i in idxs:
+            assert i < size
+
+            node = cell.nodes[i]
+
+            cell.start_node(node)
+
+    for cell in cells:
+        cell.await_started()
+
+
+def create_idle_dump_and_copy_to_log_dir(control_utility: ControlUtility, node: ClusterNode, log_dir: str) -> str:
+    """
+    Creates a dump file and copies it to the log directory.
+    :param control_utility ControlUtility.
+    :param node ClusterNode.
+    :param log_dir Path to log directory.
+    :return: Path to idle-verify dump file.
+    """
+    control_utility.idle_verify()
+
+    dump = control_utility.idle_verify_dump(node)
+
+    return copy_file_to_dest(node, dump, log_dir)
+
+
+def start_cell(test_context, config: IgniteConfiguration, num_nodes: int, jvm_opts: list) -> IgniteService:
```

Comment:
Actually I don't think that this method improve readability. Maybe you can put it within the start_cells function without docs?

Reply:
done